Added work with MSSQL instead of Dictionary

Lana-Ryzhova opened the pull request.

Lana-Ryzhova committed:
Added work with MSSQL instead of Dictionary

LinusTyr reviewed (changes requested):
TodoItems/Controller seems to have been introduced. Why?

The StorageController is probably the one you want to be using.
The DbContext has no definition of the StorageItems. This can't be working.

Furthermore the database assumes to have been locally installed. Not using Docker.

Also please add the .sln file and any other metadata files I need to open/execute the assignment.

Lana-Ryzhova committed:
chore: remove unused TodoController and TodoItem

Lana-Ryzhova committed:
Fix: removed Todo, added StorageController with EF + local SQL Server, added .sln and migrations

Lana-Ryzhova commented:
Fix assignment 4: StorageController with EF Core + local SQL Server

- Removed unused TodoController and TodoItem
- Added StorageController with EF Core integration
- Configured AppDbContext with local SQL Server (no Docker)
- Added .sln file and migrations
- Verified endpoints with Postman (POST/GET working)

Lana-Ryzhova committed:
Merge branch 'main' into feature-database

LinusTyr reviewed (changes requested):
You've solved a lot of the things.

However, the database is still not run as a container in docker. See https://github.com/SFX-IT-Orienteringskursen-VT25/setup-mssql-with-docker-example/blob/main/SetupMssqlExample/DockerStarter.cs for inspiration.

Lana-Ryzhova committed:
chore: add .env to .gitignore (do not commit secrets)

Lana-Ryzhova committed:
feat: start MSSQL in Docker for Development via DockerStarter; load .env; add migrations and solution

Lana-Ryzhova committed:
Merge branch 'feature-database' of github.com-utb:SFX-IT-Orienteringskursen-VT25/assign4-adding-database into feature-database

Lana-Ryzhova commented:
- Added Docker support to run MSSQL in development via DockerStarter
- Configured connection using .env (secrets not committed)
- Added EF Core migrations and AppDbContext
- Implemented StorageItem model and StorageController
- Updated solution and project files accordingly
- Resolved merge conflicts and updated .gitignore

LinusTyr commented on Server/AdditionApi/Program.cs:
```diff
@@ -0,0 +1,46 @@
+var port = Environment.GetEnvironmentVariable("MSSQL_PORT") ?? "1433";
+
+// Формируем строку подключения вручную
+var connectionString =
+    $"Server={host},{port};Database={dbName};User Id=sa;Password={saPassword};TrustServerCertificate=True;";
+
+// Записываем строку подключения в конфигурацию
+builder.Configuration["ConnectionStrings:DefaultConnection"] = connectionString;
```
Nice solution.

I'd set the value of ConnectionStrings:DefaultConnection in the appsettings.json to empty for clarity purposes. Just like you did in the Development.json file.

LinusTyr reviewed (approved):
Good job!

Well done.
